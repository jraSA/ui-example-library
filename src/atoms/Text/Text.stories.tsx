import React from 'react';
import { Meta } from '@storybook/react/types-6-0';
import Text from '.';

export default {
    title: 'Atoms/Text',
    component: Text,
} as Meta;

const Template = (args) => <Text {...args} />;

export const SimpleTextComponent = Template.bind({});
SimpleTextComponent.args = {
    value: 'foo-boo',
};
