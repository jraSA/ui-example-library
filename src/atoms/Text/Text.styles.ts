import styled from 'styled-components';
import BaseTheme from '../../theme/BaseTheme';
import { getThemeValue } from '../../theme/ThemeUtils';

const { color, fonts } = BaseTheme;

const { black } = color;
const { primaryFont, fontSizes } = fonts;

const TextContainer = styled.div`
  font-family: ${primaryFont}, arial, sans-serif;
  font-size: ${(props) => getThemeValue(props, 'textSize', fontSizes[6])}px;
  text-align: left;
  color: ${(props) => getThemeValue(props, 'textColor', black)};
`;

export { TextContainer };
