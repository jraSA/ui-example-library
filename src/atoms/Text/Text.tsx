import React from 'react';
import { TextContainer } from './Text.styles';

interface Props {
    value: string
};

const Text = (props: Props) => {
    const { value } = props;
    return (
        <TextContainer>
            {value}
        </TextContainer>
    )
};

export default Text;
