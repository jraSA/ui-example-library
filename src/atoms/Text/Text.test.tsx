import React from 'react';
import renderer from 'react-test-renderer';
import Text from '.';

describe('Render test', () => {
  const wrapper = renderer.create(<Text value={'foo-boo'} />);

  test('WHEN I render the component THEN it should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
