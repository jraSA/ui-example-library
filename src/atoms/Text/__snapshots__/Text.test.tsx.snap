// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render test WHEN I render the component THEN it should match the snapshot 1`] = `
<div
  className="sc-bdfBwQ RdbVo"
>
  foo-boo
</div>
`;
